frontend/src/store/auth/reducer.ts

```diff
@@ -1,25 +1,43 @@
-import { createReducer } from '@reduxjs/toolkit';
+import { createReducer, isAnyOf } from '@reduxjs/toolkit';
 
 import { DataStatus } from 'common/enums/enums';
-import { signUp } from './actions';
+import { TokenPair, UserBaseResponseDto } from 'common/types/types';
+import { logout, refreshTokens, signIn, signUp } from './actions';
 
 type State = {
   dataStatus: DataStatus;
+  error: string | undefined;
+  user: UserBaseResponseDto | null;
+  tokens: TokenPair | null;
 };
 
 const initialState: State = {
   dataStatus: DataStatus.IDLE,
+  user: null,
+  tokens: null,
```

Comment:
Redux storage it's not a good place to store auth tokens even if you use redux-persist better to use localstorage directly.

Reply:
why? isn't it more convenient not to deal with localstorage at all, and let redux-persist do it automatically? moreover, if we plan to move the app to react-native in the future, we will have to change every occurrence of localstorage to the native storage, while with redux it will all happen out of the box

Reply:
@foxt451 Please use localStorage as mentioned above